pypeit/spectrographs/spectrograph.py

```diff
@@ -1245,14 +1245,30 @@ def get_datacube_bins(self, slitlength, minmax, num_wave):
                 Number of wavelength steps.  Given by::
                     int(round((wavemax-wavemin)/delta_wave))
 
-        Args:
+        Returns:
             :obj:`tuple`: Three 1D `numpy.ndarray`_ providing the bins to use
             when constructing a histogram of the spec2d files. The elements
             are :math:`(x,y,\lambda)`.
         """
         msgs.warn("No datacube setup for spectrograph: {0:s}".format(self.name))
         return None
 
+    def flatfield_structure(self, ff_struct, gpmask):
+        r"""
+        Perform a model fit to any instrument-specific flatfield structure.
+
+        Args:
+            ff_struct (`numpy.ndarray`_):
+                An image of the flatfield structure.
+            gpmask (`numpy.ndarray`_):
+                Good pixel mask (True=good), the same shape as ff_struct.
+
+        Returns:
+            `numpy.ndarray`_: A model fit to the flatfield structure.
+        """
+        msgs.warn("Flatfield structure is not implemented for spectrograph: {0:s}".format(self.name))
```

Comment:
Not sure we want a warning that will show up every time. Maybe better to print out a message if the non-default behavior is executed for KCWI.